core/src/main/java/com/scalar/db/service/StorageModule.java

```diff
@@ -17,8 +19,8 @@ public StorageModule(DatabaseConfig config) {
 
   @Override
   protected void configure() {
-    bind(DistributedStorage.class).to(config.getStorageClass()).in(Singleton.class);
-    bind(DistributedStorageAdmin.class).to(config.getStorageAdminClass()).in(Singleton.class);
+    bind(DistributedStorage.class).to(config.getStorageClass());
```

Comment:
Also, removed the `Singleton` scope. This is because the storage and admin and transaction manager instances are closeable, so if they are closed, then Guice always returns already closed instances.